### PR TITLE
feat(Component-DataTable): Use Glamorous Relational CSS Selectors

### PR DESCRIPTION
**Pull request by medelbou**
Use Glamorous Relational CSS Selectors for more control over css rules.

### CODE REVIEW

File: src/components/DataTable.jsx [hunk 1]
```diff
@@ -183,18 +183,18 @@ export default class DataTable extends Base {
     this.props.onChange(rows);
   }
 
-  renderHeaderItem(style) {
+  renderHeaderItem() {
     const headers = this.processHeaders();
     const headerTitles = Object.keys(headers).map((header, i) => {
-      const arrowStyle = this.props.orderBy && this.props.orderBy.direction === 'asc'
-        ? style.TheadArrowUp : style.TheadArrowDown;
+      const arrowCSSClass = this.props.orderBy && this.props.orderBy.direction === 'asc'
```
Comment (ryanballa) [inline src/components/DataTable.jsx]: nice!

File: src/components/DataTable.jsx [hunk 2]
```diff
@@ -214,14 +213,12 @@ export default class DataTable extends Base {
     return this.props.multiSelectable ? [selectAllHeader, headerTitles] : headerTitles;
   }
 
-  renderHeaderItems(style) {
+  renderHeaderItems() {
     return (
       <TableRow
-        isHeader={true}
```
Comment (ryanballa) [inline src/components/DataTable.jsx]: was this not needed?

File: src/components/DataTable.jsx [hunk 3]
```diff
@@ -315,58 +312,78 @@ export default class DataTable extends Base {
...
-        padding: '20px',
-        fontWeight: 'bold',
-        textAlign: 'left',
+        '& thead': {
+          '& tr': {
+            background: clr.backgroundColors.tableHeader,
+            borderBottom: `1px solid ${clr.structuralColors.divider}`,
+            '& th': {
+              cursor: 'pointer',
+              background: clr.backgroundColors.tableHeader,
```
Comment (ryanballa) [inline src/components/DataTable.jsx]: Just an observation, we really ought to use the same linter rules, I know we force alphabetical ordering in components.

File: src/components/TableCell.jsx [hunk 4]
```diff
@@ -21,26 +20,15 @@ export default class TableCell extends Base {
...
-        border: 'none',
-        borderBottom: `1px solid ${clr.structuralColors.divider}`,
-        color: clr.textColors.primary,
-        padding: '20px',
-      },
-    };
-
-    const tdStyle = Object.assign({}, style.TBodyItems, { width: this.props.width });
-    const StyledTD = styledComonent('td', tdStyle);
+    const tdStyle = { width: this.props.width };
```
Comment (ryanballa) [inline src/components/TableCell.jsx]: does this even need to be a variable?

File: src/components/TableRow.jsx [hunk 5]
```diff
@@ -10,41 +9,32 @@ export default class TableRow extends Base {
...
-    const { isHeader, isHoverable, isSelected, style: styleFromProps } = this.props;
-
-    let rowStyle = isHoverable ? style.TableRowWithHover : {};
-    rowStyle = isSelected ? style.selected : rowStyle;
-    rowStyle = isHeader ? style.SortableTableHeader : rowStyle;
-    rowStyle = Object.assign({}, styleFromProps, rowStyle);
-
-    const Row = styledComonent('tr', rowStyle);
+    const { isHoverable, isSelected, style } = this.props;
+
```
Comment (ryanballa) [inline src/components/TableRow.jsx]: nit: remove line

**Comment by ryanballa** [comment]
oh, I'd also use a different type of commit than feature, this feels more like refactoring or a bug fix to me. It doesn't really provide any new functionality, especially if one was using Sprinkles alone.

File: src/components/DataTable.jsx (hunk 2 above)
Comment (medelbou) [inline src/components/DataTable.jsx]: no, I added that one before, but it's not needed any more, as I use relational css selectors to target thead -> tr, which I know is a header row.

File: src/components/TableCell.jsx (hunk 4 above)
Comment (medelbou) [inline src/components/TableCell.jsx]: You are right, it doesn't anymore, I'll remove it.

File: src/components/TableRow.jsx (hunk 5 above)
Comment (medelbou) [inline src/components/TableRow.jsx]: sure, I tend to like more vertical space :)

**Pull request merged**